File: algorithms/226-invert-binary-tree/invert-binary-tree.go
package problem226

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

// 递归
func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}
	left := invertTree(root.Left)
	right := invertTree(root.Right)
	root.Left = right
	root.Right = left
	return root
}

File: algorithms/226-invert-binary-tree/invert-binary-tree.py
"""
226. invert binary tree

翻转二叉树
"""


class TreeNode:
    def __init__(self, x):
        # dfs
        self.val = x
        self.left: TreeNode = None
        self.right: TreeNode = None


class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        """
        recursion

        时间复杂度：O(n)
        空间复杂度：O(n)
        """
        if not root:
            return root

        root.left, root.right = root.right, root.left
        self.invertTree(root.left)
        self.invertTree(root.right)

        return root
